extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -29,35 +64,47 @@ export class WorkspaceService implements IWorkspaceService {
 	}
 
 	async addProjectsToWorkspace(projectFiles: vscode.Uri[]): Promise<void> {
-		if (vscode.workspace.workspaceFile) {
-			const currentProjects: vscode.Uri[] = this.getProjectsInWorkspace();
-			const newWorkspaceFolders: string[] = [];
-			let newProjectFileAdded = false;
-			for (const projectFile of projectFiles) {
-				if (currentProjects.findIndex((p: vscode.Uri) => p.fsPath === projectFile.fsPath) === -1) {
-					currentProjects.push(projectFile);
-					newProjectFileAdded = true;
-
-					// if the relativePath and the original path is the same, that means the project file is not under
-					// any workspace folders, we should add the parent folder of the project file to the workspace
-					const relativePath = vscode.workspace.asRelativePath(projectFile, false);
-					if (vscode.Uri.file(relativePath).fsPath === projectFile.fsPath) {
-						newWorkspaceFolders.push(path.dirname(projectFile.path));
-					}
-				}
-			}
+		if (!projectFiles || projectFiles.length === 0) {
+			return;
+		}
 
-			if (newProjectFileAdded) {
-				// Save the new set of projects to the workspace configuration.
-				await this.setWorkspaceConfigurationValue(ProjectsConfigurationName, currentProjects.map(project => this.toRelativePath(project)));
-				this._onDidWorkspaceProjectsChange.fire();
+		// a workspace needs to be open to add projects
+		if (!vscode.workspace.workspaceFile) {
+			const result = await vscode.window.showWarningMessage(constants.CreateWorkspaceConfirmation, constants.OkButtonText);
+			if (result === constants.OkButtonText) {
+				await this.CreateNewWorkspaceForProject(projectFiles[0].fsPath);
```

Comment:
I know the code following won't be executed in real world, but I still think you should add a 'return;' after this statement and add comment here. doing this would also be useful for you to write unit test.